pytorch_lightning/utilities/argparse.py

```diff
@@ -135,29 +134,44 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
     return name_type_default
 
 
-def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
+def add_argparse_args(cls, parent_parser: ArgumentParser, *, inplace=False) -> ArgumentParser:
     r"""Extends existing argparse by default `Trainer` attributes.
 
     Args:
         cls: Lightning class
         parent_parser:
             The custom cli arguments parser, which will be extended by
             the Trainer default arguments.
+        inplace:
+            By default, this is False, and creates a new ArgumentParser connected to
+            ``parent_parser``.
+            If True, then this will add the arguments directly to the passed ``parent_parser``
+            instance. (Use this when using `add_argument_group()`.)
 
     Only arguments of the allowed types (str, float, int, bool) will
     extend the `parent_parser`.
 
     Examples:
+
+        # Option 1: Default usage.
         >>> import argparse
         >>> from pytorch_lightning import Trainer
         >>> parser = argparse.ArgumentParser()
         >>> parser = Trainer.add_argparse_args(parser)
         >>> args = parser.parse_args([])
+
+        # Option 2: Using add_argument_group().
+        >>> parser = argparse.ArgumentParser()
+        >>> parser = Trainer.add_argparse_args(parser.add_argument_group("pl.Trainer"), inplace=True)
+        >>> args = parser.parse_args([])
     """
-    parser = ArgumentParser(
-        parents=[parent_parser],
-        add_help=False,
-    )
+    if inplace:
+        parser = parent_parser
```

Comment:
Nuts. CI fails here 'cause this logic is just wrong. Will fix.

Reply:
Erg blech: b55521e5
Fixes it, but adds some nasty edge cases. Sorry :sweat_smile: 

Can a PL dev resolve this convo if they find it satisfactory? (or, well, marginally acceptable?)

Reply:
Fine with me, left some more comments